mne/decoding/xdawn.py

```diff
@@ -0,0 +1,206 @@
+"""Contains refactored Xdawn class."""
+
+# Authors: Asish Panda <asishrocks95@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from ..cov import _regularized_covariance
+from ..preprocessing.xdawn import _Xdawn
```

Comment:
@agramfort where shall we put this `xdawn.py`. Currently we have the original in preproc and the new one is a transformer.

We can either 
* regroup all transformers
* write each new transformer in the same folder/file that its function relates too (e.g. TimeFrequencyTransformer would be in mne/time_frequency)

WDYT?